### PR TITLE
Bugfix/fc-252

### PR DESCRIPTION
**Pull request by harishp8889**
# Description
Fixed a bug when the driver fails to mount the volumes when there is no iSCSI support on the node.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| https://github.com/dell/csm/issues/252 |

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [x] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Tested on Kubernetes cluster with only FC connectivity 
- [x] Tested on Kubernetes cluster with all protocol connectivity
- [x] Ran unit tests

image

### CODE REVIEW

**Review by gilltaran — APPROVED** [review]

File: pkg/controller/publisher.go [hunk 1]
```diff
@@ -167,29 +167,33 @@ func (s *SCSIPublisher) addTargetsInfoToPublishContext(
...
-		return err
+		log.Error("error unable to get NVMeFC targets from array", err)
 	}
 	for i, t := range nvmefcTargetInfo {
 		publishContext[fmt.Sprintf("%s%d", common.PublishContextNVMEFCPortalsPrefix, i)] = t.Portal
 		publishContext[fmt.Sprintf("%s%d", common.PublishContextNVMEFCTargetsPrefix, i)] = t.Target
 	}
-
+	// If the system is not capable of any protocol, then we will through the error
+	if len(iscsiTargetsInfo) == 0 && len(fcTargetsInfo) == 0 && len(nvmefcTargetInfo) == 0 {
```
Comment (spriya-m) [inline pkg/controller/publisher.go:194]: do we need to add check for nvme TCP targets also?

Comment (harishp8889) [reply pkg/controller/publisher.go:194]: I don't see any specific target information for NVMeTCP @harshitap26 please confirm whether we need it separately?

Comment (harshitap26) [reply pkg/controller/publisher.go:194]: Can we remove this condition from here and add it in the nodeStageVolume function after we get the NVMeTCP targets as well? @harishp8889

Comment (spriya-m) [reply pkg/controller/publisher.go:194]: ok

**Review by spriya-m — APPROVED** [review]

**Pull request merged**